Sources/Utility/ArgumentParser.swift

```diff
@@ -200,6 +203,26 @@ public struct PathArgument: ArgumentKind {
     public static var completion: ShellCompletion = .filename
 }
 
+public struct ExecutableName: ArgumentKind {
+    public let name: String
+
+    public init(argument: String) throws {
+        name = argument
+    }
+
+    public static var completion: ShellCompletion = .function("_swift_executable")
```

Comment:
I think it actually make sense to move it to `ArgumentProtocol` and remove the coupling from the type. That way we can add completion option when constructing an option. This will allow us to have greater flexibility as each option can then decide if they want autocompletion and of what kind.